lib/src/cpp/utils/XMLManager.hpp

```diff
@@ -66,146 +66,173 @@ class XMLManager
     ~XMLManager();
 
     /**
-     * @brief Validate the document, and save to disk if valid
+     * @brief Validate the document, and save to disk if valid.
      *
      * @throw ElementInvalid exception if document does not pass parser validation
      */
     void validate_and_save_document();
 
     /**
-     * @brief Remove the given node from the parent node
-     *
-     * @param parent_node DOMNode parent node which should contain the node to be deleted
-     * @param node_to_be_deleted DOMNode node to be deleted
+     * @brief Remove the selected node.
      *
      * @throw ElementInvalid exception if node is the last node element (could not be deleted)
      */
-    void clear_node(
-            xercesc::DOMNode*& parent_node,
-            xercesc::DOMNode*& node_to_be_deleted);
+    void clear_node();
 
     /**
-     * @brief Set the value to node object
+     * @brief Set the value to node object.
      *
-     * @param node DOMNode node to be set
-     * @param value to be set in the node
+     * @param[in] value to be set in the node
      */
     void set_value_to_node(
-            xercesc::DOMNode*& node,
             const std::string& value);
 
     /**
-     * @brief Temporal function to get the doc object
+     * @brief Set the attribute value associated to the attribute name to node object.
      *
-     * @return xercesc::DOMDocument* doc object
+     * @param[in] name to set the node attribute
+     * @param[in] value to be set in the node
      */
-    xercesc::DOMDocument* get_doc();
+    void set_attribute_to_node(
+            const std::string& name,
+            const std::string& value);
 
-    /*
-     * AUX get_node functions based on the implementation needs
+    /**
+     * @brief Set the attribute value of all node siblings to the given value associated to given name.
+     *
+     * @param[in] name to set the node attribute
+     * @param[in] value to be set in the node
      */
-    xercesc::DOMNode* get_node(
-            const std::string& tag_name);
-
-    xercesc::DOMNode* get_node(
-            const std::string& tag_name,
-            const std::string* index);
-
-    xercesc::DOMNode* get_node(
-            const std::string& tag_name,
-            const std::string& att_name,
-            const std::string& att_value);
+    void set_siblings_attribute(
+            const std::string& name,
+            const std::string& value);
 
-    xercesc::DOMNode* get_node(
-            const std::string& tag_name,
-            const std::string* index,
-            const std::string& att_name,
-            const std::string& att_value);
 
-    xercesc::DOMNode* get_node(
-            xercesc::DOMNode*& parent_node,
-            const std::string& tag_name);
+    /**
+     * @brief Get the node value (only for simple cases)
+     *
+     * @throw ElementNotFound exception if could not obtain node value
+     *
+     * @return std::string node value (Empty string if error)
```

Comment:
This is only obtained with a bad use of the api, like trying to obtain a value from a node which has not been initialized yet, or a complex node (which has child nodes but has no value)

Reply:
```std::string XMLManager::get_node_value()
{
    if (xercesc::XMLString::transcode(last_node->getNodeValue()) != nullptr)
    {
        return xercesc::XMLString::transcode(last_node->getNodeValue());
    }
    // Throw ElementNotFound exception
    std::string exception_message = "Could not find value of element ";
    exception_message += xercesc::XMLString::transcode(last_node->getNodeName());
    throw ElementNotFound(exception_message + "\n");
}
```

Looking at the implementation, I think we should remove the last part, because in case of failure an exception is thrown so the return value is not important.

```suggestion
     * @return std::string node value
```